### PR TITLE
ICU-22374 guard gCollationBinKey behind if !UCONFIG_NO_COLLATION

### PR DESCRIPTION
**Pull request by srl295**
- no other callers when this is set

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22374
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [ ] Issue accepted (done by Technical Committee after discussion)

### CODE REVIEW

**Review by richgillam — APPROVED** [review]

**Comment by srl295** [comment]
@richgillam OK to merge this and get it off the books?

File: icu4c/source/common/uresdata.cpp [hunk 1]
```diff
@@ -1040,13 +1040,15 @@ enum {
...
 static const char *const gUnknownKey="";
 
+#if !UCONFIG_NO_COLLATION
 /* resource table key for collation binaries: "%%CollationBin" */
 static const char16_t gCollationBinKey[]={
     0x25, 0x25,
     0x43, 0x6f, 0x6c, 0x6c, 0x61, 0x74, 0x69, 0x6f, 0x6e,
     0x42, 0x69, 0x6e,
     0
 };
```
Comment (markusicu) [inline icu4c/source/common/uresdata.cpp:1050]: optional:
```suggestion
// resource table key for collation binaries
static const char16_t gCollationBinKey[]=u"%%CollationBin";
```

**Review by markusicu — COMMENTED** [review]
lgtm

**Review by richgillam — DISMISSED** [review]
I like Markus's suggestion too.

**Review by FrankYFTang — DISMISSED** [review]

File: icu4c/source/common/uresdata.cpp (hunk 1 above)
Comment (srl295) [reply icu4c/source/common/uresdata.cpp:1050]: ooh - i didn't look at the data close enough I guess. Good suggestion.

**Review by echeran — APPROVED** [review]

**Review by FrankYFTang — APPROVED** [review]
LGTM

**Review by markusicu — APPROVED** [review]

**Pull request merged**